src/core/PWSfile.cpp

```diff
@@ -86,6 +86,8 @@ PWSfile::VERSION PWSfile::ReadVersion(const StringX &filename, const StringX &pa
     VERSION v;
     if (PWSfileV3::IsV3x(filename, v))
       return v;
+    else if (filename.rfind(_T(".") V4_SUFFIX) == filename.size() - wcslen(V4_SUFFIX) - 1)
+      return V40;
     else if (PWSfileV4::IsV4x(filename, passkey, v))
```

Comment:
Why can't we extend this check so there continues to be 1 if branch for each version?

Reply:
You could do the same for V3, but it is less important there since the V3 check can be done very quickly (as the header of the file reveals it). For V4 the current check is expensive, which is why I suggest the change in this PR.

Reply:
I'm only talking about v4. There are now 2 branches - one checks for extension, and the other checks the contents. Can we combine them?

Reply:
Ah, now I understand your question. To accommodate that, I think it is best to move the new "if" to the PWSfileV4::IsV4x method, would you prefer that over the current implementation? Then I'll update the PR.

Reply:
Yes, that would better, but let's close on the other thread first.